### PR TITLE
feat: use updated grid driver when needed during provisioning

### PR DESCRIPTION
**Pull request by UtheMan**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: e2e/scenario/validators.go [hunk 1]
```diff
@@ -72,6 +72,24 @@ func NvidiaSMINotInstalledValidator() *LiveVMValidator {
 	}
 }
 
+func NvidiaSMIInstalledValidator() *LiveVMValidator {
```
Comment (UtheMan) [inline e2e/scenario/validators.go:75]: I wanted to combine installed/not installed validators for smi somehow, but I think it's cleaner to just leave these like that for now.

File: pkg/agent/baker.go [hunk 2]
```diff
@@ -996,6 +999,14 @@ func gpuNeedsFabricManager(size string) bool {
 	return datamodel.FabricManagerGPUSizes[strings.ToLower(size)]
 }
 
+func getAKSGPUImageSha(size string) string {
+	if useGridDrivers(size) {
+		return datamodel.AKSGPUGridSha
```
Comment (cameronmeissner) [inline pkg/agent/baker.go:1004]: case inconsistencies :(

non-blocking

File: pkg/agent/datamodel/const.go [hunk 3]
```diff
@@ -122,6 +122,14 @@ const (
 	Nvidia510CudaDriverVersion = "cuda-510.47.03"
 	Nvidia525CudaDriverVersion = "cuda-525.85.12"
 	Nvidia510GridDriverVersion = "grid-510.73.08"
+	Nvidia535GridDriverVersion = "grid-535.54.03"
+)
+
+// These SHAs will change once we update aks-gpu images in aks-gpu repository. We do that fairly rarely at this time
```
[the remark this answers is not in the document]
Comment (UtheMan) [reply pkg/agent/datamodel/const.go:128]: Agreed, but for now its the best we have. Previously it was hardcoded directly in the script :'(

File: e2e/scenario/scenario_ubuntu2204-gpu-grid.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+import (
+	"github.com/Azure/agentbaker/pkg/agent/datamodel"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute"
+)
+
+func ubuntu2204GPUGridDriver() *Scenario {
+	return &Scenario{
+		Name:        "ubuntu2204-gpu-grid",
```
[the remark this answers is not in the document]
Comment (UtheMan) [reply e2e/scenario/scenario_ubuntu2204-gpu-grid.go:11]: We 100% should do it, we sometimes only have quota for these in certain regions + we want to add more GPU scenarios soon.

File: pkg/agent/baker.go (hunk 2 above)
Comment (UtheMan) [reply pkg/agent/baker.go:1004]: fixed

File: pkg/agent/datamodel/const.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (UtheMan) [reply pkg/agent/datamodel/const.go:128]: I can pick this up, but probably out of scope for this PR

[the remark this answers is not in the document]
Comment (cameronmeissner) [reply pkg/agent/datamodel/const.go:128]: yep, meant this as a follow-up item or some sort

**Review by cameronmeissner — APPROVED** [review]

**Pull request merged**